generators/example/src/main/scala/Generator.scala

```diff
@@ -24,3 +25,34 @@ object Generator extends GeneratorApp {
   generateTestSuiteMakefrags
   generateArtefacts
 }
+
+object GeneratorWithHwacha extends GeneratorApp {
+  // add unique test suites
+  override def addTestSuites {
+    implicit val p: Parameters = params
+    TestSuiteHelper.addRocketTestSuites
+    TestSuiteHelper.addBoomTestSuites
+
+    // add hwacha bmarks + asm tests
+    import hwacha.HwachaTestSuites._
+    TestGeneration.addSuites(rv64uv.map(_("p")))
+    TestGeneration.addSuites(rv64uv.map(_("vp")))
+    TestGeneration.addSuite(rv64sv("p"))
+    TestGeneration.addSuite(hwachaBmarks)
+  }
+
+  override def generateTestSuiteMakefrags {
+    addTestSuites
+    var frag = TestGeneration.generateMakefrag + "\nSRC_EXTENSION = $(base_dir)/hwacha/$(src_path)/*.scala" + "\nDISASM_EXTENSION = --extension=hwacha"
```

Comment:
Need to double check this line... but I think this breaks in Chipyard since `hwacha` is under the `generators` section and not in the root of the repo.